ui/kotlinx-coroutines-android/README.md

```diff
@@ -5,6 +5,63 @@ Provides `Dispatchers.Main` context for Android applications.
 Read [Guide to UI programming with coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/coroutines-guide-ui.md)
 for tutorial on this module.
 
+# Optimization
+
+You can optimize the size of the coroutines library in Android projects by statically removing 
+debugging and other advanced features using R8 by adding the following rule to your 
+`proguard-rules.pro` file:
+
+```
+# Statically turn off all debugging facilities and assertions
+-assumenosideeffects class kotlinx.coroutines.DebugKt {
+    boolean getASSERTIONS_ENABLED() return false;
+    boolean getDEBUG() return false;
+    boolean getRECOVER_STACK_TRACES() return false;
+}
+
+-checkdiscard class kotlinx.coroutines.DebugKt
+-checkdiscard class kotlinx.coroutines.internal.StackTraceRecoveryKt
+
+# Use simpler and smaller CommonPool for both Dispatchers.Default and IO
+-assumenosideeffects class kotlinx.coroutines.DispatchersKt {
+    boolean getDEFAULT_SCHEDULER() return false;
+    boolean getIO_SCHEDULER() return false;
+}
+
+-checkdiscard class kotlinx.coroutines.DispatchersKt
+-checkdiscard class kotlinx.coroutines.scheduling.CoroutinesScheduler
+
+# Remove the custom, fast service loader implementation
+-assumevalues class kotlinx.coroutines.internal.MainDispatcherLoader {
```

Comment:
we should add a comment to make sure only R8 users enable this, as ProGuard doesn't have the necessary ServiceLoader optimization

I'm also currently working on a mechanism for AGP 3.6.0 to enable putting seperate rules for PG and R8 in META-INF, so we'll be able to do this automatically for anyone on R8 1.5+ or R8 1.6+ (depending on which one we choose)